hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1298,7 +1300,8 @@ private static Option<Schema> tryResolveSchemaForTable(HoodieTableMetaClient dat
       TableSchemaResolver schemaResolver = new TableSchemaResolver(dataTableMetaClient);
       return Option.of(schemaResolver.getTableAvroSchema());
     } catch (Exception e) {
-      throw new HoodieException("Failed to get latest columns for " + dataTableMetaClient.getBasePath(), e);
+      LOG.warn("Failed to get latest columns for {}", dataTableMetaClient.getBasePath());
+      return Option.empty();
```

Comment:
Should this still throw the exception?  If there are completed commits, the table schema is present; any unexpected error should fail the job instead of returning an empty schema and empty list of columns, potentially causing the index to skip those columns unintentionally.

Reply:
Makes sense. I don't recall why i changed to a warn log. Probably, I was debugging something. I have reverted now.